src/hooks/use-create-editor.tsx

```diff
@@ -117,7 +117,14 @@ export const useCreateEditor = () => {
     modelMap,
     getManuscript: () => manuscript,
     getCurrentUser: () => user,
-    setComment: (comment?: CommentAnnotation) => dispatch({ comment }),
+    setComment: (comment?: CommentAnnotation) => {
+      if (comment) {
+        const state = getState()
+        const contribution = buildContribution(state.user._id)
+        comment.contributions = [contribution]
+        dispatch({ newComments: state.newComments.set(comment._id, comment) })
```

Comment:
Why do we need to mutate existing comments map? Can we create a new one so if there is something depending on an update of it, it will be able to react?

Reply:
updated